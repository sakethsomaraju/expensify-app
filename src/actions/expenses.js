import uuid from 'uuid'

//add expense

export const addExpense = ({description='',note='',amount=0,createdAt=0}={})=>{
    return{
        type:'ADD_EXPENSE',
        expense:{
            id:uuid(),
            description,
            note,
            amount,
            createdAt
        }
    }
   }
   // remove Expense
   export const removeExpense = ({id}={})=>{
       return{
           type:'REMOVE_EXPENSE',
           id
       }
   }
   //edit expense
  export const editExpense = (id,updates)=>{
       return{
           type:'EDIT_EXPENSE',
           id,
           updates
       }
   
   }