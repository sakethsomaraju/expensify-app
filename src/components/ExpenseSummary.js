import React from 'react'
import {connect} from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import SelectExpensesTotal from'../selectors/expenses-total'


export const ExpenseSummary = ({expenseCount,expenseTotal}) =>{
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formatedExpensesTotal = numeral(expenseTotal/100).format('$0,0.00')
    return(
        <div>
        <h1>Veiwing {expenseCount} {expenseWord} totalling {formatedExpensesTotal}</h1>
        </div>
    )
}

const mapStateToProps = (state)=>{
    const visibleExpenses = selectExpenses(state.expenses,state.filters)

    return{
        expenseCount: visibleExpenses.length,
        expenseTotal:SelectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary)