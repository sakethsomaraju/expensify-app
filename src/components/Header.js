import React from 'react' 
import {NavLink} from 'react-router-dom'

const Header = ()=>{
    return (
        <header>
            <h1>expensify</h1>
            <NavLink to="/" activeClassName="is-active" exact={true}>home </NavLink>
            <NavLink to="/create" activeClassName="is-active">create expense </NavLink>
            <NavLink to="/help" activeClassName="is-active">help </NavLink>

        </header>
    )
}

export default Header