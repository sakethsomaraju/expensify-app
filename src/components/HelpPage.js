import React from 'react';


const HelpPage =()=>{
    return(
    <div>help page is !
    </div>)
}

export default HelpPage