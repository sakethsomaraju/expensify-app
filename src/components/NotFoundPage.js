import React from 'react';
import {Link} from 'react-router-dom'

const NotFoundPage =()=>{
    return(
    <div><h1>404!</h1> <Link to="/">go home !</Link>
    </div>)
}
export default NotFoundPage