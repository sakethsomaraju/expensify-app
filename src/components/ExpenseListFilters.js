import React from 'react'
import {connect} from 'react-redux'
import {setTextFilter,sortByDate,sortByAmount, setStartDate,setEndDate} from '../actions/filters'
import {DateRangePicker} from 'react-dates'

export class ExpenseListFilters extends React.Component{
    state={
        calenderFocussed:null
    }
    onDatesChange =({startDate,endDate})=>{
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange=(calenderFocussed)=>{
            this.setState(()=>({calenderFocussed}))
    }
    onTextChange = (e)=>{
        this.props.setTextFilter(e.target.value)
    }
    onSortChange=(e)=>{
        if(e.target.value==='date'){
            this.props.sortByDate()
        }else if(e.target.value==='amount'){
            this.props.sortByAmount()
        }
    }
    render(){
        return(
            <div>
                <input type="text" value={this.props.filters.text} onChange={this.onTextChange}></input>
        
                <select 
                    value={this.props.filters.sortBy}
                    onChange={this.onSortChange}
                >
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calenderFocussed}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={()=> false}
                    showClearDates={true}
                />
            </div>
            )

    }
}



const mapStateToProps = (state)=>{
    return{
        filters:state.filters
    }

}
const mapDispatchToProps = (dispatch)=>{
    return {
        setTextFilter:(text)=>dispatch(setTextFilter(text)),
        setStartDate:(startDate)=>dispatch(setStartDate(startDate)),
        setEndDate:(endDate)=>dispatch(setEndDate(endDate)),
        sortByDate:()=>dispatch(sortByDate),
        sortByAmount:()=>dispatch(sortByAmount)
    }
}

export default connect (mapStateToProps,mapDispatchToProps)(ExpenseListFilters)