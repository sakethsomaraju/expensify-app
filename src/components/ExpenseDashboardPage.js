import React from 'react';
import ConnectedExpenseList from './ExpenseList'
import ExpenseListFlters from './ExpenseListFilters'
import ExpenseListFilters from './ExpenseListFilters';


const ExpenseDashboardPage =()=>{
    return(
    <div>
        <ExpenseListFilters />
        <ConnectedExpenseList />
    </div>)
}

export default ExpenseDashboardPage