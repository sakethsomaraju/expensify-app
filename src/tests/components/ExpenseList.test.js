import React from 'react'
import {shallow} from 'enzyme'
import {ExpenseList} from '../../components/ExpenseList'
import expenses from '../fixtures/expenses'
import toJSON from 'enzyme-to-json'

test('should remember Expenselist with expenses',()=>{
    const wrapper = shallow(<ExpenseList expenses={expenses} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
})

test('Should render expense list with empty message',()=>{
    const wrapper = shallow(<ExpenseList expenses={[]} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
})