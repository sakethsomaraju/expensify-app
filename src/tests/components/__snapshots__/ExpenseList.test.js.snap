// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense list with empty message 1`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;

exports[`should remember Expenselist with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gun"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={1935}
    createdAt={-345600000}
    description="rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={5}
    createdAt={345600000}
    description="Credit card"
    id="3"
    key="3"
    note=""
  />
</div>
`;
